File: solutions/opus-1/opus/python/anagrams.py
import sys


def main():
    groups = {}
    for line in sys.stdin.read().split():
        groups.setdefault("".join(sorted(line)), []).append(line)
    result = sorted(sorted(words) for words in groups.values())
    sys.stdout.write("".join(" ".join(g) + "\n" for g in result))


main()

File: solutions/opus-1/opus/ruby/anagrams.rb
words = STDIN.read.split
groups = words.group_by { |w| w.chars.sort.join }.values.map(&:sort)
groups.sort_by! { |g| g[0] }
out = groups.map { |g| g.join(" ") }.join("\n")
STDOUT.write(out.empty? ? "" : out + "\n")
